addons/point_of_sale/static/src/js/Chrome.js

```diff
@@ -201,6 +201,13 @@ odoo.define('point_of_sale.Chrome', function(require) {
         }
 
         setupBarcodeParser() {
+            if (!this.env.pos.company.nomenclature_id) {
+                const error = new Error("Missing barcode nomenclature");
+                console.error(error, error.stack);
+                error.stack = `The barcode nomenclature setting is not configured. \
+                Make sure to configure it on your Point of Sale configuration settings`;
+                throw error;
```

Comment:
I was just thinking of `throw new Error("Missing barcode nomenclature");`.

How about:

1. `throw new Error("Missing barcode nomenclature", { cause: { kind: "SETUP_BARCODE_ERROR", message: "<message here>" } })`.
2. in `Chrome.start`, explicitly handle this error. Show any `ErropPopup` you want and use the information from error.cause.

I know it's more code but you won't need to replace "stack" props in the error.

Reply:
And probably in the future, the "catch" clause in the `Chrome.start` can be standardize to handle different kinds of errors.